weave-js/src/components/PagePanelComponents/Home/Browse3/pages/PlaygroundPage/usePlaygroundState.ts

```diff
@@ -97,23 +133,16 @@ export const usePlaygroundState = () => {
             }
           }
         }
-        if (inputs.n) {
-          newState.nTimes = parseInt(inputs.n, 10);
-        }
-        if (inputs.temperature) {
-          newState.temperature = parseFloat(inputs.temperature);
-        }
         if (inputs.response_format) {
           newState.responseFormat = inputs.response_format.type;
         }
-        if (inputs.top_p) {
-          newState.topP = parseFloat(inputs.top_p);
-        }
-        if (inputs.frequency_penalty) {
-          newState.frequencyPenalty = parseFloat(inputs.frequency_penalty);
-        }
-        if (inputs.presence_penalty) {
-          newState.presencePenalty = parseFloat(inputs.presence_penalty);
+        for (const [key, value] of Object.entries(NUMERIC_SETTINGS_MAPPING)) {
+          if (inputs[value.pythonValue]) {
```

Comment:
Should this be checking if the value is undefined? if the value is a true zero I think you'd want to set that